app/src/main/java/at/aau/serg/activities/LoginActivity.kt

```diff
@@ -0,0 +1,84 @@
+package at.aau.serg.activities
+
+import android.content.Context
+import android.content.ContextWrapper
+import android.content.Intent
+import android.content.SharedPreferences
+import android.os.Bundle
+import android.view.View
+import android.widget.EditText
+import android.widget.Toast
+import androidx.activity.enableEdgeToEdge
+import androidx.annotation.UiThread
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import at.aau.serg.MainActivity
+import at.aau.serg.R
+import at.aau.serg.logic.Authentication
+import at.aau.serg.logic.StoreToken
+import com.google.gson.Gson
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.Response
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.IOException
+import java.security.KeyStore
+
+class LoginActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        enableEdgeToEdge()
+        setContentView(R.layout.activity_login)
+        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
+            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
+            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
+            insets
+        }
+    }
+
+    fun btnLogin_Clicked(view: View) {
+        val username = findViewById<EditText>(R.id.editTextUsername).text
```

Comment:
Can we please add some validation of the user input (Username is not empty, Password is not empty, etc.)

Reply:
the user input validation is handled in Authentication.kt

Reply:
Yes, but it is not shown in the `LoginActivity`